provider/dynamodb.go

```diff
@@ -281,12 +281,16 @@ func (store *dynamodbOnlineStore) CheckHealth() (bool, error) {
 		return false, fferr.NewExecutionError(pt.DynamoDBOnline.String(), err)
 	}
 	if len(listOutput.TableNames) == 0 {
-		return false, NewProviderError(Connection, pt.DynamoDBOnline, Ping, "no tables found")
+		wrapped := fferr.NewConnectionError(pt.DynamoDBOnline.String(), fmt.Errorf("no tables found"))
```

Comment:
Little confusing. Why does 0 tables mean it was a connection error?

Reply:
As discussed, we'll change this check to see if the user has the correct permissions and remove the list tables check.